src/core/index.ts

```diff
@@ -417,6 +417,24 @@ export class ZarrArray {
     }
   }
 
+  public async getRawChunk(chunkCoords: number[]): Promise<RawArray> {
+    if (chunkCoords.length !== this.shape.length) {
+      throw new Error(`Chunk coordinates ${chunkCoords.join(".")} do not correspond to shape ${this.shape}.`);
+    }
+    for (let i = 0; i < chunkCoords.length; i++) {
+      const coordIndex = chunkCoords[i];
+      const maxCoordIndex = Math.ceil(this.shape[i] / this.chunks[i]) - 1;
+      if (coordIndex > maxCoordIndex) {
```

Comment:
or coordIndex < 0, it doesn't support looping around

I propose we change this for-loop to calls to [this function](https://github.com/gzuidhof/zarr.js/blob/master/src/core/indexing.ts#L110), it will check the bounds, handle loop around, and throw an error that can be `instanceof`'ed.

Reply:
fixed.